examples/EmailerAgent/emailer/agent.py

```diff
@@ -0,0 +1,419 @@
+# the United States Government. Neither the United States Government nor the
+# United States Department of Energy, nor Battelle, nor any of their
+# employees, nor any jurisdiction or organization that has cooperated in the
+# development of these materials, makes any warranty, express or
+# implied, or assumes any legal liability or responsibility for the accuracy,
+# completeness, or usefulness or any information, apparatus, product,
+# software, or process disclosed, or represents that its use would not infringe
+# privately owned rights. Reference herein to any specific commercial product,
+# process, or service by trade name, trademark, manufacturer, or otherwise
+# does not necessarily constitute or imply its endorsement, recommendation, or
+# favoring by the United States Government or any agency thereof, or
+# Battelle Memorial Institute. The views and opinions of authors expressed
+# herein do not necessarily state or reflect those of the
+# United States Government or any agency thereof.
+#
+# PACIFIC NORTHWEST NATIONAL LABORATORY operated by
+# BATTELLE for the UNITED STATES DEPARTMENT OF ENERGY
+# under Contract DE-AC05-76RL01830
+# }}}
+
+from __future__ import absolute_import, print_function
+
+from collections import defaultdict
+
+# Import the email modules we'll need
+from email.mime.text import MIMEText
+import logging
+import socket
+
+# Import smtplib for the actual sending function
+import smtplib
+import sys
+
+import gevent
+from volttron.platform.agent.utils import get_utc_seconds_from_epoch
+from volttron.platform.vip.agent import Agent, Core, PubSub, compat
+from volttron.platform.agent import utils
+from volttron.platform.messaging import topics
+from volttron.platform.messaging.health import ALERT_KEY, STATUS_BAD, Status, \
+    STATUS_GOOD
+from volttron.platform.vip.agent import Agent
+
+utils.setup_logging()
+_log = logging.getLogger(__name__)
+__version__ = '1.3.1'
+
+"""
+The `pyclass:EmailAgent` is responsible for sending emails for an instance.  It
+has been written so that any agent on the instance can send emails through it
+via the `pymethod:send_email` method or through the pubsub message bus (see
+`topics.PLATFORM_SEND_EMAIL`).
+
+A default configuration for this agent is as follows.
+
+.. code-block:: json
+    {
+        "smtp_address": "smtp.foo.com",
+        "from_address": "billy@foo.com",
+        "to_addresses=to_address": ["ann@foo.com", "bob@gmail.com"],
+        "allow_frequency_minutes": 10
+    }
+
+By default any alerts will be sent through this agent.  In addition all emails
+will be published to the record/sent_email topic for a historian to be able
+to capture that data.
+"""
+
+
+class EmailerAgent(Agent):
+
+    def __init__(self, config_path, **kwargs):
+        super(EmailerAgent, self).__init__(**kwargs)
+
+        config = utils.load_config(config_path)
+        smtp_address = config.get("smtp-address", None)
```

Comment:
These should be self. member variables and after validation these will be set in the configure_main function